crates/ruff/src/rules/flake8_comprehensions/rules/unnecessary_comprehension.rs

```diff
@@ -33,7 +34,7 @@ impl AlwaysAutofixableViolation for UnnecessaryComprehension {
 pub fn unnecessary_comprehension(
     checker: &mut Checker,
     expr: &Expr,
-    elt: &Expr,
+    elts: &[&Expr],
```

Comment:
I'm tempted to say that the change in signature here means we should use two separate functions entirely -- what do you think?

Reply:
Actually, I did start by defining a separate function but then most of the code remained the same. So, then I thought to combine them. Separate functions do have a pro of having a simpler body to understand and will be the same as the reference implementation. 

Now that I think of it, post reading your previous comment as well, I believe separate functions will be better to avoid any implicit expectations.